CompilerForCAP/gap/ProofAssistant.gi

```diff
@@ -17,3 +17,171 @@ InstallGlobalFunction( CapJitDisableProofAssistantMode, function ( )
+
+# literal booleans together with `and` and `or`
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "true and value",
+        dst_template := "value",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "value and true",
+        dst_template := "value",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "false and value",
+        dst_template := "false",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "value and false",
+        dst_template := "false",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "true or value",
+        dst_template := "true",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "value or true",
+        dst_template := "true",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "false or value",
+        dst_template := "value",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "value" ],
+        src_template := "value or false",
+        dst_template := "value",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+# not true => false
+# Note: GAP simplifies `not true` to `false` during parsing, so we have to specify src_template_tree directly
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ ],
+        src_template := "not true",
+        src_template_tree := rec(
+            type := "EXPR_NOT",
+            op := rec(
+                type := "EXPR_TRUE",
+            ),
+        ),
+        dst_template := "false",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+# not false => true
+# Note: GAP simplifies `not false` to `true` during parsing, so we have to specify src_template_tree directly
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ ],
+        src_template := "not false",
+        src_template_tree := rec(
+            type := "EXPR_NOT",
+            op := rec(
+                type := "EXPR_FALSE",
+            ),
+        ),
+        dst_template := "true",
+        apply_in_proof_assistant_mode := "only",
+    )
+);
+
+# ForAll( list, l -> true ) => true
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "list" ],
+        src_template := "ForAll( list, l -> true )",
+        dst_template := "true",
+        apply_in_proof_assistant_mode := "only",
```

Comment:
Why is this logic template, for example, only restricted to the proof assistant?

Reply:
Because I do not want to slow down the "normal" compilation by logic templates which typically only appear in proofs. If any of these templates also appears more widespread in "normal" compilation, we can of course turn it into a regular logic template.

Reply:
Thanks :)